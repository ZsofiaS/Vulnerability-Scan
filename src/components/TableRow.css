.mainRow {
  border-top: 1px solid gainsboro;
}
.secondaryRow {
  text-align: justify;
}
.infoButton {
  border: none;
  background-color: white;
}
.infoButton:hover {
  cursor: pointer;
}
.warning {
  padding: 0.3rem;
}
.critical {
  background-color: fireBrick;
}
.high {
  background-color: orangeRed;
}
.medium {
  background-color: orange;
}
.low {
  background-color: yellow;
}
.information {
  background-color: #3498DB;
}
