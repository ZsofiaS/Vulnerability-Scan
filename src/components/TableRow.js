import React from 'react';

const TableRow = (props) => {
  return (
    <tr>
      <td>{props.severity}</td>
      <td>{props.name}</td>
      {/*
      <td>{props.description}</td>
      <td>{props.solution}</td>
      <td>{props.references}</td>
      */}
      <td>{props.cvssBaseScore}</td>
      <td>{props.renderAssets}</td>
      <td>More info</td>
    </tr>
  )
}

export default TableRow;
