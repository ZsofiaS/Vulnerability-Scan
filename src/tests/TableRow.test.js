import React from 'react';
import { shallow, mount } from 'enzyme';
import TableRow from '../components/TableRow';
import toJson from 'enzyme-to-json';

const issues = [
  {
    severity: 'moderate'
  }
]

describe('renders TableRow component', () => {
  it('renders without crashing', () => {
    shallow(<TableRow />);
  })
});

describe('it passes props', () => {
  const tableRow = mount(<TableRow severity={issues[0].severity} />);
  it('accepts props', () => {
    expect(tableRow.props().severity).toEqual('moderate');
  })
})

describe('snapshot', () => {
  it('TableRow snapshot', () => {
    const tree = shallow(<TableRow />);
    expect(toJson(tree)).toMatchSnapshot();
  })
})
