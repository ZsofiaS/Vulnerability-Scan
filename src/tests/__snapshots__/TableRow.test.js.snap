// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot TableRow snapshot 1`] = `
<Fragment>
  <tr>
    <td
      className="mainRow"
    >
      <p
        className="warning undefined"
      />
    </td>
    <td
      className="mainRow"
    />
    <td
      className="mainRow"
    />
    <td
      className="mainRow"
    />
    <td
      className="mainRow"
    >
      <p
        className="infoButton"
        onClick={[Function]}
      >
        ...
      </p>
    </td>
  </tr>
</Fragment>
`;
